xml_loader/shift_generation.py

```diff
@@ -0,0 +1,269 @@
+from gurobipy import *
+from pathlib import Path
+from xml_loader.xml_loader import *
+import xml.etree.ElementTree as ET
+
+
+def get_time_steps(root):
+    demands = get_demand_definitions(root)
+    time_step_length = 1
+    for demand in demands:
+        for i in range(len(demand.end)):
+            if(demand.end[i] - int(demand.end[i]) > 0):
+                if(demand.end[i] - int(demand.end[i]) == 0.5 and time_step_length > 0.5):
+                    time_step_length = 0.5
+                elif(demand.end[i] - int(demand.end[i]) in [0.25,0.75] and time_step_length > 0.25):
+                    time_step_length = 0.25
+                # TODO: Check if elif-sentence below makes sense. What if we have one event at 17.80 and the rest on
+                #  whole hours? We would get 0.8 that should give time steps in minutes but that is not < 0.25.
```

Comment:
Tror dette vil bli en bug ja! Tanker @hkgrov ?

Reply:
Stemmer nok det. Er det slik at dette er blitt fikset? Hvis det er fikset, i hvilken branch har det blitt fikset?

Reply:
Ikke fikset. Tenker at det kommer i backlog for neste uke siden det er så special case. Viktigere å få denne koden gjennom nå :)